cmds/core/yes/yes_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2017-2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"bufio"
+	"testing"
+
+	"github.com/u-root/u-root/pkg/testutil"
+)
+
+func TestMain(m *testing.M) {
+	testutil.Run(m, main)
+}
+
+func TestYes(t *testing.T) {
+	cmd := testutil.Command(t)
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		t.Logf("Could not return a pipe that will be connected to the command's standard output, received err: %v", err)
+	}
+
+	if err := cmd.Start(); err != nil {
+		t.Log(err)
+	}
+
+	in := bufio.NewScanner(stdout)
+
+	for i := 0; i < 1000; i++ {
+		if in.Scan() && in.Text() != "y" {
```

Comment:
If in.Scan returns false, you just loop 1000 times without checking anything, where you should instead be failing if it fails to scan.

Something like this:
if !in.Scan() {
    t.Fatalf(//some error message)
}
if text := in.Text(); text != "y" {
    t.Errorf(//some error message)
}

Reply:
That is also a very good point. I need to test that condition first just to make sure I am not looping through 1000 times even if it does not scan anything! Thank you, Gan.